src/Ingest.php

```diff
@@ -94,11 +95,11 @@ private function createStream()
     /**
      * @param  resource  $stream
      */
-    private function sendPayload($stream, string $payload): void
+    private function sendPayload($stream, Payload $payload): void
     {
         $written = 0;
-        $remainingPayload = $payload;
-        $payloadLength = strlen($payload);
+        $remainingPayload = $payload->pull();
+        $payloadLength = strlen($remainingPayload);
```

Comment:
This is a substantial memory improvement.

Previously, we would be passing around a string. Although we trim the string down in this function, it original string would still be in scope in parent functions.

Now that we pass around and object and the string is resolved within this function, as soon as we trim the string down in size, the original string may be garbage collected.